<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>bugphyzz • bugphyzz</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="bugphyzz">
<meta property="og:description" content="bugphyzz">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bugphyzz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/bugphyzz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/checks.html">Curation Checks</a>
    </li>
    <li>
      <a href="../articles/descriptive_statistics.html">Descriptive statistcs of taxa annotations</a>
    </li>
    <li>
      <a href="../articles/write_physiology_sets.html">Write physiology sets</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>bugphyzz</h1>
            
      
      
      <div class="hidden name"><code>bugphyzz.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">bugphyzz</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://purrr.tidyverse.org" class="external-link">purrr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This document shows the main functions and operations that can be
used to work with the bugphyzz datasets, such as seeing all
physiologies, look at specific physiologies, and make signature sets.
Bugphyzz is a collection of physiological features about bacteria. We
are curating the databases through 3 main methods: webscraping, bulk
download from online databases, and manual curation from literature.
Mainly this database will be used to analyze microbiome data.</p>
</div>
<div class="section level2">
<h2 id="definitions">Definitions<a class="anchor" aria-label="anchor" href="#definitions"></a>
</h2>
<ul>
<li>
<strong>NCBI ID</strong>: National Center for Bioinfomatic numeric
ID to link to specific taxa.</li>
<li>
<strong>Genome ID</strong>: From NCBI, another type of numeric ID,
that links to a specific sequenced genome. Not all taxa have a NCBI ID,
and might only have a Genome ID</li>
<li>
<strong>Accession ID</strong>: The accession number is a unique
identifier assigned to a record in sequence databases such as GenBank.
Several NCBI databases use the format [alphabetical prefix][series of
digits]. A change in the record in some databases (e.g. GenBank) is
tracked by an integer extension of the accession number, an
Accession.</li>
<li>
<strong>Taxon name</strong>: a human readable organismal name that
can delinate which level taxa the annotation is describing.</li>
<li>
<strong>Attribute</strong>: a value for a physiology, that is mapped
to an ontology whenever possible. These are defined in the
<code>attribute</code> column of [inst/extdata/attributes.tsv](<a href="https://github.com/waldronlab/bugphyzz/blob/main/inst/extdata/attributes.tsv" class="external-link uri">https://github.com/waldronlab/bugphyzz/blob/main/inst/extdata/attributes.tsv</a>
<strong>Physiology</strong>: a microbial characteristic that can be
measured or observed, that should be mapped to an ontology wherever
possible. These are defined in <a href="https://github.com/waldronlab/bugphyzz/blob/main/inst/extdata/links.tsv" class="external-link">inst/extdata/links.tsv</a>.</li>
<li>
<strong>Attribute value</strong>: a value describing a physiology.
Allowable values are defined by the <code>validity</code> column of <a href="https://github.com/waldronlab/bugphyzz/blob/main/inst/extdata/attributes.tsv" class="external-link">inst/extdata/attributes.tsv</a>.
For categorical attributes (such as “obligately aerobic”), values can be
“TRUE” and “FALSE”, for example. For numeric attributes, these can be
numeric values.</li>
<li>
<strong>Attribute source</strong>: a citeable domain which the
annotation was obtained. A source can be another database or primary
literature.</li>
<li>
<strong>Frequency</strong>: Describes how often the attritbute
(phenotype) occurs: Rarely, Sometimes, Usually, Always, or Unknown. If
the source makes the attribute unclear how often it occurs, then an
unknown is given.</li>
<li>
<strong>Evidence codes</strong>: As the Gene Ontology defines
different forms of evidence a source was obtained the annotation.</li>
</ul>
<p>EXP: Experimental data, the attribute was obtained through labortory
confirmations, and ideally more than once.</p>
<p>IGC: Computational predicted through means of metagenomic analysis,
or more broadly, genome analysis.</p>
<p>ASR: Anstrestal State Reconstruction, was predicted using ASR
computional techniques to predicted up and down the taxonmic tree of
taxa that has a value.</p>
<ul>
<li>
<strong>Note</strong>: Any additional information that does not fit
in any fixed column. The information here is variable.</li>
<li>
<strong>MIDAS ID</strong>: Located in the Habitat attribute. Any
annotation from the MIDAS database is assigned an ID number. We provided
that ID here.</li>
<li>
<strong>Parent name</strong>: The parent of descendant given in
Taxon name column. Values can indudes names or NA.</li>
<li>
<strong>Species name</strong>: The species name, or descendant of
parent, given in the Taxon name column. Values can indudes names or
NA.</li>
<li>
<strong>Unit</strong>: defines unit of attribute, like µm.</li>
<li>
<strong>Rank</strong>: The taxonomic rank of the annotation. Values
can indudes names or NA.</li>
<li>
<strong>Parent Rank</strong>: The taxonmic rank of the parent of the
annotation. Values can indudes names or NA.</li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of bugphyzz with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"waldronlab/bugphyzz"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="some-examples-of-how-to-create-signatures">Some examples of how to create signatures<a class="anchor" aria-label="anchor" href="#some-examples-of-how-to-create-signatures"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">bugphyzz</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://purrr.tidyverse.org" class="external-link">purrr</a></span><span class="op">)</span></span>
<span>    <span class="va">aer</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/physiologies.html">physiologies</a></span><span class="op">(</span><span class="st">'aerophilicity'</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">ph</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/physiologies.html">physiologies</a></span><span class="op">(</span><span class="st">'optimal ph'</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Example 1</span></span>
<span><span class="co">## Create a signature of genera with experimental evidence at a </span></span>
<span><span class="co">## Frequency of always using the NCBI taxids. Minimumn of 10 microbes per</span></span>
<span><span class="co">## signature</span></span>
<span><span class="va">aer_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignatures.html">makeSignatures</a></span><span class="op">(</span></span>
<span>    <span class="va">aer</span>, tax.id.type <span class="op">=</span> <span class="st">'NCBI_ID'</span>, tax.level <span class="op">=</span> <span class="st">'genus'</span>, </span>
<span>    Frequency <span class="op">=</span> <span class="st">'always'</span>, Evidence <span class="op">=</span> <span class="st">'exp'</span>, min.size <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Dropping 3648 rows with FALSE values.</span></span>
<span><span class="co">#&gt; Creating signature for categorical attributes.</span></span>
<span></span>
<span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_int</a></span><span class="op">(</span><span class="va">aer_sig</span>, <span class="va">length</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 aerobic               anaerobic facultatively anaerobic </span></span>
<span><span class="co">#&gt;                     574                     294                     174 </span></span>
<span><span class="co">#&gt;         microaerophilic      obligately aerobic    obligately anaerobic </span></span>
<span><span class="co">#&gt;                      17                      52                     104</span></span>
<span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">aer_sig</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.x</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; $aerobic</span></span>
<span><span class="co">#&gt; [1]  291967 1078830</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $anaerobic</span></span>
<span><span class="co">#&gt; [1] 1427378   33951</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`facultatively anaerobic`</span></span>
<span><span class="co">#&gt; [1]  2147 12914</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $microaerophilic</span></span>
<span><span class="co">#&gt; [1] 46352 12960</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`obligately aerobic`</span></span>
<span><span class="co">#&gt; [1] 442869    434</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`obligately anaerobic`</span></span>
<span><span class="co">#&gt; [1] 258514  31980</span></span>
<span></span>
<span></span>
<span><span class="co">## Example 2</span></span>
<span><span class="co">## Create a signature of acidiphilic species using the Taxon names</span></span>
<span><span class="va">acid_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignatures.html">makeSignatures</a></span><span class="op">(</span></span>
<span>    df <span class="op">=</span> <span class="va">ph</span>, tax.id.type <span class="op">=</span> <span class="st">'Taxon_name'</span>, tax.level <span class="op">=</span> <span class="st">'species'</span>,</span>
<span>    max <span class="op">=</span> <span class="fl">4</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Creating signature for continuous attribute.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">acid_sig</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Leptospirillum ferriphilum"     "Sulfobacillus acidophilus"     </span></span>
<span><span class="co">#&gt; [3] "Acidithiobacillus ferrivorans"  "Acidithiobacillus ferrooxidans"</span></span>
<span><span class="co">#&gt; [5] "Acidimicrobium ferrooxidans"    "Thermoplasma volcanium"</span></span>
<span></span>
<span><span class="co">## Create a signature of Alkaliphilic species using the Taxon names</span></span>
<span><span class="va">basic_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignatures.html">makeSignatures</a></span><span class="op">(</span></span>
<span>    df <span class="op">=</span> <span class="va">ph</span>, tax.id.type <span class="op">=</span> <span class="st">'Taxon_name'</span>, tax.level <span class="op">=</span> <span class="st">'species'</span>, </span>
<span>    min <span class="op">=</span> <span class="fl">8</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Creating signature for continuous attribute.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">basic_sig</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Arhodomonas aquaeolei"        "Glaciecola nitratireducens"  </span></span>
<span><span class="co">#&gt; [3] "Pseudomonas coronafaciens"    "Sodalis glossinidius"        </span></span>
<span><span class="co">#&gt; [5] "Photorhabdus luminescens"     "Hyphomicrobium nitrativorans"</span></span></code></pre></div>
<p><sup>Created on 2022-09-01 with <a href="https://reprex.tidyverse.org" class="external-link">reprex v2.0.2</a></sup></p>
</div>
<div class="section level2">
<h2 id="bugphyzzanalyses">bugphyzzAnalyses<a class="anchor" aria-label="anchor" href="#bugphyzzanalyses"></a>
</h2>
<p>Some analyses using data provided by this package can be found at <a href="https://waldronlab.io/bugphyzzAnalyses/" class="uri">https://waldronlab.io/bugphyzzAnalyses/</a></p>
</div>
<div class="section level2">
<h2 id="how-to-add-a-new-attribute-in-github-to-the-bugphyzz-database">How to add a new attribute in GitHub to the BugPhyzz database<a class="anchor" aria-label="anchor" href="#how-to-add-a-new-attribute-in-github-to-the-bugphyzz-database"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Add attributes of physiology to Attributes using ontology terms</li>
<li>Add reference(s) to Confidence in Curation</li>
<li>Add Google Doc “publish to web” link and add in Links</li>
<li>Add Google Doc link to Source Link</li>
</ol>
</div>
<div class="section level2">
<h2 id="installation-1">Installation<a class="anchor" aria-label="anchor" href="#installation-1"></a>
</h2>
<p>Bugphyzz can be installed with:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"waldronlab/bugphyzz"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="importing-datasets">Importing datasets<a class="anchor" aria-label="anchor" href="#importing-datasets"></a>
</h2>
<p>The main function for importing datasets is the
<code>physiologies</code> function, which has a single argument,
<code>keyword</code>, and returns a list of data.frames. The
<code>keyword</code> argument accepts a character vector specifying the
name(s) of the dataset(s) that will be imported.</p>
<p>A datset with categorical attribute values:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aer</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="fu"><a href="../reference/physiologies.html">physiologies</a></span><span class="op">(</span><span class="st">"aerophilicity"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">aer</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 11,472 × 16</span></span></span>
<span><span class="co">#&gt;    NCBI_ID Genom…¹ Acces…² Taxon…³ Attri…⁴ Attri…⁵ Attri…⁶ Evide…⁷ Frequ…⁸ Note </span></span>
<span><span class="co">#&gt;      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>  <span style="text-decoration: underline;">442</span>869 <span style="color: #BB0000;">NA</span>      <span style="color: #949494;">""</span>      Acanth… obliga… TRUE    Bergey… exp     always  <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>  <span style="text-decoration: underline;">291</span>967 <span style="color: #BB0000;">NA</span>      <span style="color: #949494;">""</span>      Acaric… aerobic TRUE    Bergey… exp     always  <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>  <span style="text-decoration: underline;">291</span>968 <span style="color: #BB0000;">NA</span>      <span style="color: #949494;">""</span>      Acaric… aerobic TRUE    https:… unknown always  <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>  <span style="text-decoration: underline;">329</span>726 <span style="color: #BB0000;">NA</span>      <span style="color: #949494;">"</span>NC_00… Acaryo… aerobic TRUE    BacMap  unknown always  <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>  <span style="text-decoration: underline;">258</span>514 <span style="color: #BB0000;">NA</span>      <span style="color: #949494;">""</span>      Acetan… obliga… TRUE    Bergey… exp     always  <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 1<span style="text-decoration: underline;">427</span>378 <span style="color: #BB0000;">NA</span>      <span style="color: #949494;">""</span>      Acetat… anaero… TRUE    Bergey… exp     always  <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>   <span style="text-decoration: underline;">31</span>980 <span style="color: #BB0000;">NA</span>      <span style="color: #949494;">""</span>      Acetit… obliga… TRUE    Bergey… exp     always  <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>   <span style="text-decoration: underline;">35</span>829 <span style="color: #BB0000;">NA</span>      <span style="color: #949494;">""</span>      Acetiv… obliga… TRUE    Bergey… exp     always  <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>  <span style="text-decoration: underline;">186</span>831 <span style="color: #BB0000;">NA</span>      <span style="color: #949494;">""</span>      Acetoa… obliga… TRUE    Bergey… exp     always  <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>     434 <span style="color: #BB0000;">NA</span>      <span style="color: #949494;">""</span>      Acetob… obliga… TRUE    Bergey… exp     always  <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 11,462 more rows, 6 more variables: Strain &lt;lgl&gt;, Rank &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Parent_name &lt;chr&gt;, Parent_NCBI_ID &lt;int&gt;, Parent_rank &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Confidence_in_curation &lt;chr&gt;, and abbreviated variable names ¹​Genome_ID,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​Accession_ID, ³​Taxon_name, ⁴​Attribute, ⁵​Attribute_value,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁶​Attribute_source, ⁷​Evidence, ⁸​Frequency</span></span></span></code></pre></div>
<p>A dataset with numeric attribute values:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ph</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="fu"><a href="../reference/physiologies.html">physiologies</a></span><span class="op">(</span><span class="st">'optimal ph'</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; Finished optimal ph</span></span>
<span><span class="va">ph</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 886 × 14</span></span></span>
<span><span class="co">#&gt;    NCBI_ID Genom…¹ Acces…² Taxon…³ Attri…⁴ Attri…⁵ Attri…⁶ Evide…⁷ Frequ…⁸ Rank </span></span>
<span><span class="co">#&gt;      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>  <span style="text-decoration: underline;">244</span>596 <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Hoefle… optima…    7    The Mi… exp     usually spec…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 1<span style="text-decoration: underline;">125</span>964 <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Loktan… optima…    6    The Mi… exp     usually spec…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>  <span style="text-decoration: underline;">178</span>606 <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Leptos… optima…    1.65 The Mi… exp     usually spec…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>   <span style="text-decoration: underline;">53</span>633 <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Sulfob… optima…    1.8  The Mi… exp     usually spec…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>  <span style="text-decoration: underline;">160</span>808 <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Acidit… optima…    1.9  The Mi… exp     usually spec…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     920 <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Acidit… optima…    2    The Mi… exp     usually spec…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>   <span style="text-decoration: underline;">53</span>635 <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Acidim… optima…    2    The Mi… exp     usually spec…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>   <span style="text-decoration: underline;">50</span>339 <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Thermo… optima…    2    The Mi… exp     usually spec…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>   <span style="text-decoration: underline;">28</span>034 <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Sulfob… optima…    2.15 The Mi… exp     usually spec…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>   <span style="text-decoration: underline;">67</span>780 <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Edward… optima…    2.5  The Mi… exp     usually spec…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 876 more rows, 4 more variables: Parent_name &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Parent_NCBI_ID &lt;int&gt;, Parent_rank &lt;chr&gt;, Confidence_in_curation &lt;chr&gt;, and</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   abbreviated variable names ¹​Genome_ID, ²​Accession_ID, ³​Taxon_name,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁴​Attribute, ⁵​Attribute_value, ⁶​Attribute_source, ⁷​Evidence, ⁸​Frequency</span></span></span></code></pre></div>
<p>The <code>physiologiesList</code> function can be used to display
which datasets are available through the <code>physiologies</code>
function before importing them:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/physiologiesList.html">physiologiesList</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "acetate producing"                    </span></span>
<span><span class="co">#&gt;  [2] "aerophilicity"                        </span></span>
<span><span class="co">#&gt;  [3] "animal pathogen"                      </span></span>
<span><span class="co">#&gt;  [4] "antimicrobial resistance"             </span></span>
<span><span class="co">#&gt;  [5] "antimicrobial sensitivity"            </span></span>
<span><span class="co">#&gt;  [6] "arrangement"                          </span></span>
<span><span class="co">#&gt;  [7] "biofilm forming"                      </span></span>
<span><span class="co">#&gt;  [8] "butyrate producing"                   </span></span>
<span><span class="co">#&gt;  [9] "COGEM pathogenicity rating"           </span></span>
<span><span class="co">#&gt; [10] "disease association"                  </span></span>
<span><span class="co">#&gt; [11] "extreme environment"                  </span></span>
<span><span class="co">#&gt; [12] "gram stain"                           </span></span>
<span><span class="co">#&gt; [13] "growth medium"                        </span></span>
<span><span class="co">#&gt; [14] "growth temperature"                   </span></span>
<span><span class="co">#&gt; [15] "habitat"                              </span></span>
<span><span class="co">#&gt; [16] "health associated"                    </span></span>
<span><span class="co">#&gt; [17] "hydrogen gas producing"               </span></span>
<span><span class="co">#&gt; [18] "isolation site"                       </span></span>
<span><span class="co">#&gt; [19] "lactate producing"                    </span></span>
<span><span class="co">#&gt; [20] "length"                               </span></span>
<span><span class="co">#&gt; [21] "mutation rate per site per generation"</span></span>
<span><span class="co">#&gt; [22] "mutation rates per site per year"     </span></span>
<span><span class="co">#&gt; [23] "optimal ph"                           </span></span>
<span><span class="co">#&gt; [24] "plant pathogenicity"                  </span></span>
<span><span class="co">#&gt; [25] "shape"                                </span></span>
<span><span class="co">#&gt; [26] "spore shape"                          </span></span>
<span><span class="co">#&gt; [27] "width"                                </span></span>
<span><span class="co">#&gt; [28] "genome size"                          </span></span>
<span><span class="co">#&gt; [29] "coding genes"                         </span></span>
<span><span class="co">#&gt; [30] "sphingolipid producing"</span></span></code></pre></div>
<p>You can use the <code>keyword = "all"</code> to import all of the
datasets:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Import all of the datasets available through the physiologies function</span></span>
<span><span class="va">phys</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="../reference/physiologies.html">physiologies</a></span><span class="op">(</span>keyword <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span>, <span class="va">as_tibble</span><span class="op">)</span></span></code></pre></div>
<p>A separate function is provided to import the fatty acid compositions
dataset, <code>fattyAcidComposition</code>. This function takes no
arguments and returns a single data.frame object:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fac</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="fu"><a href="../reference/fattyAcidComposition.html">fattyAcidComposition</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fac</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 138,496 × 15</span></span></span>
<span><span class="co">#&gt;    NCBI_ID Genome_ID Accession…¹ Taxon…² Attri…³ Attri…⁴ Attri…⁵ Evide…⁶ Frequ…⁷</span></span>
<span><span class="co">#&gt;      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     562 <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          Escher… Fatty …       0 Microb… EXP     unknown</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     562 <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          Escher… Fatty …       0 Microb… EXP     unknown</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>     562 <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          Escher… Fatty …       0 Microb… EXP     unknown</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>     562 <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          Escher… Fatty …       0 Microb… EXP     unknown</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>     562 <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          Escher… Fatty …       0 Microb… EXP     unknown</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     562 <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          Escher… Fatty …       0 Microb… EXP     unknown</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>     562 <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          Escher… Fatty …       0 Microb… EXP     unknown</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>     562 <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          Escher… Fatty …       0 Microb… EXP     unknown</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>     562 <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          Escher… Fatty …       0 Microb… EXP     unknown</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>     562 <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          Escher… Fatty …       0 Microb… EXP     unknown</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 138,486 more rows, 6 more variables: Attribute_new &lt;chr&gt;, Rank &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Parent_name &lt;chr&gt;, Parent_NCBI_ID &lt;int&gt;, Parent_rank &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Confidence_in_curation &lt;chr&gt;, and abbreviated variable names ¹​Accession_ID,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​Taxon_name, ³​Attribute, ⁴​Attribute_value, ⁵​Attribute_source, ⁶​Evidence,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁷​Frequency</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="creating-microbial-signatures">Creating microbial signatures<a class="anchor" aria-label="anchor" href="#creating-microbial-signatures"></a>
</h2>
<p>Bugphyzz provides the <code>makeSignatures</code> function to create
a list of microbial signatures from a bugphyzz dataset.</p>
<div class="section level3">
<h3 id="create-signatures-of-an-attribute-with-categoricallogical-values">Create signatures of an attribute with categorical/logical
values:<a class="anchor" aria-label="anchor" href="#create-signatures-of-an-attribute-with-categoricallogical-values"></a>
</h3>
<p>Create a list of signatures with the NCBI taxids of microbes at the
taxonomic rank of genus. Only include microbes annotated based on
experimental data with a high frequency value. The minimum number of
microbes per signature is 10.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aer_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignatures.html">makeSignatures</a></span><span class="op">(</span></span>
<span>    <span class="va">aer</span>, tax.id.type <span class="op">=</span> <span class="st">'NCBI_ID'</span>, tax.level <span class="op">=</span> <span class="st">'genus'</span>, </span>
<span>    Frequency <span class="op">=</span> <span class="st">'always'</span>, Evidence <span class="op">=</span> <span class="st">'exp'</span>, min.size <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Dropping 3648 rows with FALSE values.</span></span>
<span><span class="co">#&gt; Creating signature for categorical attributes.</span></span>
<span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">aer_sig</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.x</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; $aerobic</span></span>
<span><span class="co">#&gt; [1]  291967 1078830</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $anaerobic</span></span>
<span><span class="co">#&gt; [1] 1427378   33951</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`facultatively anaerobic`</span></span>
<span><span class="co">#&gt; [1]  2147 12914</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $microaerophilic</span></span>
<span><span class="co">#&gt; [1] 46352 12960</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`obligately aerobic`</span></span>
<span><span class="co">#&gt; [1] 442869    434</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`obligately anaerobic`</span></span>
<span><span class="co">#&gt; [1] 258514  31980</span></span></code></pre></div>
<p>Number of microbes per signature:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_int</a></span><span class="op">(</span><span class="va">aer_sig</span>, <span class="va">length</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 aerobic               anaerobic facultatively anaerobic </span></span>
<span><span class="co">#&gt;                     574                     294                     174 </span></span>
<span><span class="co">#&gt;         microaerophilic      obligately aerobic    obligately anaerobic </span></span>
<span><span class="co">#&gt;                      17                      52                     104</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="create-signatures-of-an-atttribute-with-numerical-values">Create signatures of an atttribute with numerical values<a class="anchor" aria-label="anchor" href="#create-signatures-of-an-atttribute-with-numerical-values"></a>
</h3>
<p>Create a signature list with species of acidiphilic microbes (ph
&lt;= 4):</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">acid_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignatures.html">makeSignatures</a></span><span class="op">(</span></span>
<span>    df <span class="op">=</span> <span class="va">ph</span>, tax.id.type <span class="op">=</span> <span class="st">'Taxon_name'</span>, tax.level <span class="op">=</span> <span class="st">'species'</span>,</span>
<span>    max <span class="op">=</span> <span class="fl">4</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Creating signature for continuous attribute.</span></span>
<span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">acid_sig</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.x</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`optimal ph`</span></span>
<span><span class="co">#&gt; [1] "Leptospirillum ferriphilum" "Sulfobacillus acidophilus"</span></span></code></pre></div>
<p>Number of microbes:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_int</a></span><span class="op">(</span><span class="va">acid_sig</span>, <span class="va">length</span><span class="op">)</span></span>
<span><span class="co">#&gt; optimal ph </span></span>
<span><span class="co">#&gt;         13</span></span></code></pre></div>
<p>Create a signature list with species of alkaliphilic microbes (ph
&gt;= 8):</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">basic_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignatures.html">makeSignatures</a></span><span class="op">(</span></span>
<span>    df <span class="op">=</span> <span class="va">ph</span>, tax.id.type <span class="op">=</span> <span class="st">'Taxon_name'</span>, tax.level <span class="op">=</span> <span class="st">'species'</span>, </span>
<span>    min <span class="op">=</span> <span class="fl">8</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Creating signature for continuous attribute.</span></span>
<span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">basic_sig</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.x</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`optimal ph`</span></span>
<span><span class="co">#&gt; [1] "Arhodomonas aquaeolei"      "Glaciecola nitratireducens"</span></span></code></pre></div>
<p>Number of microbes:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_int</a></span><span class="op">(</span><span class="va">basic_sig</span>, <span class="va">length</span><span class="op">)</span></span>
<span><span class="co">#&gt; optimal ph </span></span>
<span><span class="co">#&gt;        138</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information:<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] tibble_3.1.8     purrr_0.3.4      bugphyzz_0.0.1.3</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] Rcpp_1.0.9        ape_5.6-2         lattice_0.20-45   tidyr_1.2.0      </span></span>
<span><span class="co">#&gt;  [5] zoo_1.8-10        assertthat_0.2.1  rprojroot_2.0.3   digest_0.6.29    </span></span>
<span><span class="co">#&gt;  [9] foreach_1.5.2     utf8_1.2.2        R6_2.5.1          plyr_1.8.7       </span></span>
<span><span class="co">#&gt; [13] RSQLite_2.2.16    evaluate_0.16     pillar_1.8.1      rlang_1.0.5      </span></span>
<span><span class="co">#&gt; [17] curl_4.3.2        uuid_1.1-0        data.table_1.14.2 taxize_0.9.100   </span></span>
<span><span class="co">#&gt; [21] blob_1.2.3        jquerylib_0.1.4   rmarkdown_2.16    pkgdown_2.0.6    </span></span>
<span><span class="co">#&gt; [25] textshaping_0.3.6 desc_1.4.1        readr_2.1.2       stringr_1.4.1    </span></span>
<span><span class="co">#&gt; [29] bit_4.0.4         compiler_4.2.1    xfun_0.32         pkgconfig_2.0.3  </span></span>
<span><span class="co">#&gt; [33] systemfonts_1.0.4 conditionz_0.1.0  htmltools_0.5.3   tidyselect_1.1.2 </span></span>
<span><span class="co">#&gt; [37] httpcode_0.3.0    mgsub_1.7.3       codetools_0.2-18  reshape_0.8.9    </span></span>
<span><span class="co">#&gt; [41] fansi_1.0.3       tzdb_0.3.0        hoardr_0.5.2      dbplyr_2.2.1     </span></span>
<span><span class="co">#&gt; [45] crayon_1.5.1      dplyr_1.0.10      rappdirs_0.3.3    crul_1.3         </span></span>
<span><span class="co">#&gt; [49] grid_4.2.1        nlme_3.1-159      jsonlite_1.8.0    lifecycle_1.0.1  </span></span>
<span><span class="co">#&gt; [53] DBI_1.1.3         magrittr_2.0.3    taxizedb_0.3.0    vroom_1.5.7      </span></span>
<span><span class="co">#&gt; [57] cli_3.3.0         stringi_1.7.8     cachem_1.0.6      fs_1.5.2         </span></span>
<span><span class="co">#&gt; [61] xml2_1.3.3        bslib_0.4.0       ellipsis_0.3.2    ragg_1.2.2       </span></span>
<span><span class="co">#&gt; [65] generics_0.1.3    vctrs_0.4.1       iterators_1.0.14  tools_4.2.1      </span></span>
<span><span class="co">#&gt; [69] bold_1.2.0        bit64_4.0.5       glue_1.6.2        hms_1.1.2        </span></span>
<span><span class="co">#&gt; [73] parallel_4.2.1    fastmap_1.1.0     yaml_2.3.5        memoise_2.0.1    </span></span>
<span><span class="co">#&gt; [77] knitr_1.40        sass_0.4.2</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
